src/BizHawk.Common/BizHawk.Common.csproj

```diff
@@ -14,8 +14,6 @@
 		<PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="5.0.0" />
 	</ItemGroup>
 	<ItemGroup>
-		<!-- target in Version project generates VersionInfo.gen.cs; note that the Version project MUST BE FIRST in solution file -->
-		<ProjectReference Include="$(ProjectDir)../BizHawk.Version/BizHawk.Version.csproj" ReferenceOutputAssembly="false" />
-		<Compile Update="VersionInfo.gen.cs" DependentUpon="VersionInfo.cs" />
+		<Analyzer Include="$(MSBuildProjectDirectory)/../../References/BizHawk.SrcGen.VersionInfo.dll" />

```

Comment:
Can be private

Reply:
This has the disadvantage of not actually using the source generator in the IDE, just on build.
Also, how would this fix the issue about project directory resolving inside of the generator?

Reply:
`<ProjectReference/>` Analyzers don't run in VS? That sounds like a bug.

Git doesn't need to know where `.git` is, the CWD just needs to be set to any subdir.